### PR TITLE
Add the Dataflow Framework tool

### PR DESCRIPTION
**Pull request by mernst**
* [x] I have not changed the `README.md` directly.

### CODE REVIEW

File: data/tools/dataflow-framework.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: Dataflow Framework
+categories:
+  - linter
+tags:
+  - java
+license: GPL with Classpath exception / MIT License
+types:
+  - cli
+source: 'https://github.com/typetools/checker-framework'
+homepage: 'https://checkerframework.org/manual/checker-framework-dataflow-manual.pdf'
```
Comment (mre) [inline data/tools/dataflow-framework.yml:10]: Not sure if we should use a pdf as the homepage. We could use the same url as for source and add the pdf as a resource maybe?

Comment (mre) [inline data/tools/dataflow-framework.yml:10]:
```suggestion
homepage: 'https://github.com/typetools/checker-framework'
```

File: data/tools/dataflow-framework.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+description: An industrial-strength dataflow framework for Java. The Dataflow Framework is used in the Checker Framework, Google’s Error Prone, Uber’s NullAway, Meta’s Nullsafe, and in other contexts.  It is distributed with the Checker Framework.
```
Comment (mre) [inline data/tools/dataflow-framework.yml:11]:
```suggestion
description: An industrial-strength dataflow framework for Java. The Dataflow Framework is used in the Checker Framework, Google’s Error Prone, Uber’s NullAway, Meta’s Nullsafe, and in other contexts.  It is distributed with the Checker Framework.
resources:
  - title: "User Manual"
    url: https://checkerframework.org/manual/checker-framework-dataflow-manual.pdf
```

**Comment by mernst** [comment]
Thanks!

**Pull request merged**